api/core/storage/organization_storage.py

```diff
@@ -42,6 +42,8 @@ async def add_credits_to_tenant(self, tenant: str, credits: float) -> None: ...
 
     async def decrement_credits(self, tenant: str, credits: float) -> TenantData: ...
 
+    async def get_organization_by_slack_channel_id(self, slack_channel_id: str) -> TenantData | None: ...
```

Comment:
Do you need the full organization object ? I would move to the PublicOrganizationStorage and return a`PublicOrganizationData`.

Reply:
I use _get_public_ord in https://github.com/WorkflowAI/WorkflowAI/pull/176/commits/63e63a9ee9b79e05667a321b73e4b8ff7632dd33